### PR TITLE
Strongly typed waf config

### PR DESCRIPTION
**Pull request by kdorosh**
cc @EItanya 

Updated enterprise code on a branch to work with the API updates in this PR to ensure they are reasonable changes
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1191

### CODE REVIEW

**Comment by kdorosh** [comment]
/kick
```
Step #14 - "build-test-assets": Head https://gcr.io/v2/solo-public/discovery/blobs/sha256:618b06fb1c327210076bf6b3ba7034109d3b46e02c9859c46505b7c03235387e: net/http: TLS handshake timeout
```

File: projects/gloo/api/v1/plugins.proto [hunk 1]
```diff
@@ -65,6 +65,9 @@ message HttpListenerPlugins {
     hcm.plugins.gloo.solo.io.HttpConnectionManagerSettings http_connection_manager_settings = 2;
     // Deprecated: Opaque config for Gloo plugins
     Extensions extensions = 3 [deprecated=true];
+    // Enterprise-only: Config for Web Application Firewall (WAF), supporting
+    // the popular ModSecurity 3.0.3 ruleset
```
Comment (marcogschmidt) [inline projects/gloo/api/v1/plugins.proto]: Super nit: do we have to be so specific with the version? If we don't have a constraint on that exact version, I think we should just call it `ModSecurity 3.0` [like they do](https://modsecurity.org).

Comment (kdorosh) [reply projects/gloo/api/v1/plugins.proto]: sure, will update

**Comment by kdorosh** [comment]
/kick

Step #14 - "build-test-assets": Put https://gcr.io/v2/solo-public/ingress/blobs/uploads/ABTmro42QA6wc6wGYw4j8aj_0-iboRp30ZhB4MGS6xf3GKLRKa83AFckBAg207aU9pYz1jSZN5L3ZZ03eZcRj_OKVU5BG_mRF37x-a_uU-Ctu47-Qr_IL0rUsudDFYOukpGle1tdAUtnMy2ggroQnmnvI89BV-hLH9lQToFuDQD0iea-FlfUiUeKWdow?digest=sha256%3A9737f4754281d36a2829a4e75a1535bfdf0570efd555f30547a16390943cc832: dial tcp 74.125.20.82:443: i/o timeout

**Pull request merged**